agent/textwatcher.go

```diff
@@ -0,0 +1,194 @@
+package agent
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"hash/crc32"
+)
+
+type TextWatcher interface {
+	Start()
+	Stop()
+	Contents() ([]byte, error)
+	Path() string
+	Kind() string
+	MarshalJSON() ([]byte, error)
+}
+
+type textWatcher struct {
+	sync.Mutex
+	keepPolling  bool
+	kind         string
+	path         string
+	UpdatedAt    time.Time
+	BeingWatched bool
+	OnChange     ChangeHandler
+	Checksum     uint32
+	CmdName      string
+	CmdArgs      []string
+	contents     func() ([]byte, error)
+	pollSleep    time.Duration
+}
+
+// File watchers track changes in the contents of a file
+func NewFileWatcher(path string, callback ChangeHandler) Watcher {
+	var kind string
+	filename := filepath.Base(path)
+	switch filename {
```

Comment:
Not sure what the first part of this refers to. The second part we have been discussing - I added and then removed these items from the payload. I can easily add them back if we want to update them with every newly received process watcher payload.